### PR TITLE
Add settings for model averaging in the frontend

### PR DESCRIPTION
**Pull request by Dref360**
Resolve #Nothing

## Description:

This PR allows the user to see the performance after bayesian model averaging. This generally improves the performance by ~1-2% and the ECE by a couple percentage as well.

In my company, we use BMA in production to get good ECE, this is why I need this toggle.

Lot of code, but mostly poutine here are the relevant bits:

* Added `model_averaging_available` to Dataset Info
* Added new startup tasks when model averaging is enabled.
* Added new query state in the FE and new toggle
    * I acknowledge that this takes a lot of space, I'm open to an alternative.

image

## Checklist:

You should check all boxes before the PR is ready. If a box does not apply, check it to acknowledge it.

* [x] **ISSUE NUMBER.** You linked the issue number (Ex: Resolve #XXX).
* [x] **PRE-COMMIT.** You ran pre-commit on all commits, or else, you
  ran `pre-commit run --all-files` at the end.
* [x] **USER CHANGES.** The changes are added to CHANGELOG.md and the documentation, if they impact
  our users.
* [x] **DEV CHANGES.**
    * Update the documentation if this PR changes how to develop/launch on the app.
    * Update the `README` files and our wiki for any big design decisions, if relevant.
    * Add unit tests, docstrings, typing and comments for complex sections.

### CODE REVIEW

**Review by JosephMarinier — APPROVED** [review]
Cool! That looks good to me!

**Comment by Dref360** [comment]
Cool! Can you trigger the CI and then y'all can merge whenever?

**Comment by JosephMarinier** [comment]
Oh yeah... 🤔 I don't know what is going on with the CI. I'll try to investigate.

**Comment by JosephMarinier** [comment]
This super elaborate suggestion from Stack Overflow (close + reopen) didn't work. Does anyone have an idea what is going on?

**Comment by istandre** [comment]
@JosephMarinier My guess would be that in your project settings you have enabled [branch protection rules](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches). For a collaborator to be able to push in main, both the backend and frontend CI need to succeed.

Also, my understanding is that workflows on PR from outside contributors might not run automatically, and might be approved first. After being approved, the workflows will get triggered.

However, both jobs only run on `push`. See [pythonci.yml#L4](https://github.com/ServiceNow/azimuth/blob/34081048a4de3900ca29ac0d37bce7026113382c/.github/workflows/pythonci.yml#L4) and [webapp_ci.yml#L4](https://github.com/ServiceNow/azimuth/blob/34081048a4de3900ca29ac0d37bce7026113382c/.github/workflows/webapp_ci.yml#L4). Since this PR comes from a fork, no push events have triggered the workflows and upon approval, workflows did not start since they only do for `push` events.

Assuming that this is reason of the issue, I think that adding the [`pull_request` event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) to the workflow files might fix this issue. You might also want to add the `pull_request_target` event to trigger workflows (which will run in the context of the main branch) regardless of approval status.

For the latter, you can have a look at [Controlling changes from forks to workflows in public repositories](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#controlling-changes-from-forks-to-workflows-in-public-repositories).

**Comment by istandre** [comment]
I have created a bug report https://github.com/ServiceNow/azimuth/issues/621 and a PR https://github.com/ServiceNow/azimuth/pull/622 to address the CI checks issue.

File: azimuth/modules/model_performance/confidence_binning.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
 class ConfidenceHistogramModule(FilterableModule[ModelContractConfig]):
     """Return a confidence histogram of the predictions."""
 
+
```
Comment (JosephMarinier) [inline azimuth/modules/model_performance/confidence_binning.py:25]: As per mypy's desire
```suggestion
```

File: azimuth/routers/app.py [hunk 2]
```diff
@@ -104,6 +104,8 @@ def get_dataset_info(
             eval=eval_dm and eval_dm.num_rows, train=training_dm and training_dm.num_rows
         ),
         similarity_available=similarity_available(config),
+        model_averaging_available=config.uncertainty is not None
+        and config.uncertainty.iterations > 1,
```
Comment (JosephMarinier) [inline azimuth/routers/app.py:108]: I think the `test_get_dataset_info` has to be adapted for that.

**Pull request merged**